projects/js-packages/publicize-components/src/components/social-post-control/index.js

```diff
@@ -1,16 +1,17 @@
 import { ThemeProvider } from '@automattic/jetpack-components';
-import { CheckboxControl } from '@wordpress/components';
+import { CheckboxControl, Disabled } from '@wordpress/components';
+import { useCallback, Fragment } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
-import { useCallback } from 'react';
 import useAttachedMedia from '../../hooks/use-attached-media';
 import styles from './styles.module.scss';
 
 /**
  * The component that handles the social post checkbox.
  *
+ * @param {boolean} disabled -  Indicates whether the checkbox is disabled or not.
  * @returns {object} The SocialPostControl Component.
  */
-export default function SocialPostControl() {
+export default function SocialPostControl( { disabled = false } ) {
```

Comment:
We don't necessarily need a default value here, do we?

Reply:
Yeah true, I thought it's easier to read like this, but I can remove that, what do you think?

Reply:
No, nothing big. It's fine either way.